src/main/java/io/supertokens/emailpassword/PasswordHashing.java

```diff
@@ -89,22 +91,39 @@ public String createHashWithSalt(String password) {
         T op();
     }
 
-    private <T> T withConcurrencyLimited(Func<T> func) {
+    private <T> T withArgon2HashingConcurrencyLimited(Func<T> func) {
         Object waiter = new Object();
         try {
-            while (!this.boundedQueue.contains(waiter)) {
+            while (!this.argon2BoundedQueue.contains(waiter)) {
                 try {
                     // put will wait for there to be an empty slot in the queue and return
                     // only when there is a slot for waiter
-                    this.boundedQueue.put(waiter);
+                    this.argon2BoundedQueue.put(waiter);
                 } catch (InterruptedException ignored) {
                 }
             }
 
             return func.op();
 
         } finally {
-            this.boundedQueue.remove(waiter);
+            this.argon2BoundedQueue.remove(waiter);
+        }
+    }
+
+    private <T> T withFirebaseSCryptConcurrencyLimited(Func<T> func) {
+        Object waiter = new Object();
+        try {
+            while (!this.firebaseSCryptBoundedQueue.contains(waiter)) {
+                try {
+                    // put will wait for there to be an empty slot in the queue and return
+                    // only when there is a slot for waiter
+                    this.firebaseSCryptBoundedQueue.put(waiter);
+                } catch (InterruptedException ignored) {
+                }
+            }
+            return func.op();
+        } finally {
+            this.argon2BoundedQueue.remove(waiter);
```

Comment:
- why argon2 here??
- Why have you made two functions for the same thing?? Just pass in the queue as an arg to one common function.